tensorflow/core/kernels/quantized_deconv_ops.cc

```diff
@@ -0,0 +1,315 @@
+                for (int f = 0; f < input_depth; f++) {
+                  const T1 input_value = input_data[
+                    (batch * input_height * input_width * input_depth) +
+                    (h * input_width * input_depth) +
+                    (w * input_depth) +
+                    (f)
+                  ];
+
+                  const T2 filter_value = filter_data[
+                    (kx * filter_width * output_depth * input_depth) +
+                    (ky * output_depth * input_depth) +
+                    (c * input_depth) +
+                    (f)
+                  ];
+
+                  total += input_value * filter_value;
+                }
+                
+                output_data[
+                  (batch * output_height * output_width * output_depth) +
+                  (ox * output_width * output_depth) +
+                  (oy * output_depth) +
+                  (c)
+                ] += total;
+
+              }
+            }
+
+
+          }
+
+        }
+      }
+    }
+
+  }
+};
+
+template <class T1, class T2, class T3,
+         template <class TF1, class TF2, class TF3> class ConvTransFunctor>
+class QuantizedDeconv2DOp: public OpKernel {
+  public:
+    explicit QuantizedDeconv2DOp(OpKernelConstruction *context)
+      : OpKernel(context) {
+      //
+      // Assertions for Attr
+
+      // Assertions for strides
+      OP_REQUIRES_OK(context, context->GetAttr("strides", &strides_));
+      OP_REQUIRES(context, strides_.size() == 4,
+                  errors::InvalidArgument("Sliding window strides field must specify 4 dimensions"));
+      OP_REQUIRES(context, strides_[1] == strides_[2],
+                  errors::InvalidArgument("Current implementation only supports equal strides in the row and column dimensions"));
+      //OP_REQUIRES(context, (strides_[1] == 1) & (strides_[2] == 1),
+      //            errors::InvalidArgument("Current implementation doesn't support stride larger than 1"));
+      OP_REQUIRES(context, (strides_[0] == 1) | (strides_[3] == 1),
+                  errors::InvalidArgument("Current implementation doesn't support stride in batch or depth dimension"));
+      OP_REQUIRES_OK(context, context->GetAttr("padding", &padding_));
+      OP_REQUIRES(context, padding_ == VALID,
+                  errors::InvalidArgument("Only VALID padding is supported in the current implementation"));
+
+      // These checkers are copied from the Conv2DFastBackpropInputOp
+      // See: conv_grad_input_ops.cc
+      // TODO: Assertions for data_format
+      // Currently QuantizedConv2DTransposeOp doesn't support this parameter, could be implemented with
+      // GetConvnetDataFormatAttrString()
+
+      // The following checkers are set for the current implementation,
+      // which are much stricter than the expected final release.
+      // TODO: Remove these checkers when they are ready
+
+      // Stride can only be 1 at the moment
+      // OP_REQUIRES(context, strides_[1] == 1,
```

Comment:
Can you clarify whether this is still true? If not, remove the commented-out code, otherwise make it active.